src/components/file/file.service.ts

```diff
@@ -239,15 +252,19 @@ export class FileService {
       uploadId,
     });
 
-    const mimeType = upload.ContentType ?? 'application/octet-stream';
+    const mimeType =
+      (tempUpload as PromiseFulfilledResult<HeadObjectOutput>).value
```

Comment:
Can `tempUpload` & `existingUpload` be resolved to `upload` in block above to not clutter this logic?